### ISSUE TITLE
Support item deletion

### ISSUE THREAD

**Issue by nodomain** [zcutlip/pyonepassword#46]
As discussed in #45 it would be awesome if delete is supported to fulfill the "update" use case.

Directly updating existing items is not needed.

**Comment by zcutlip** [comment]
Happy to work on this. What sort of timeline are you on? I think it'll be fairly straightforward, but I'd like to polish up & release item creation (#36) first.

**Comment by nodomain** [comment]
There is no huge pressure since I can workaround with manually deleting old entries after new ones have been created. However, the more elegant way of course would be to do this via API.

**Comment by zcutlip** [comment]
Okay, good to know, thanks.

In a pinch you could always spelunk into private API (at your own risk):

```python
from pyonepassword import OP
from pyonepassword._op_cli_argv import _OPArgv
from pyonepassword.api.exceptions import OPCmdFailedException


def do_delete(item_name_or_id: str):
    # Note, if you use item name rather than ID,
    # be prepared for the case where there's a name collision
    # Not a problem with item IDs which are unique
    sub_cmd_args = [item_name_or_id]
    delete_argv = _OPArgv.item_generic_argv("op", "delete", sub_cmd_args)

    op = OP()
    try:
        op._run(delete_argv)
    except OPCmdFailedException as e:
        print(e.err_output)
        # time for plan B because something went wrong

```

**Comment by zcutlip** [comment]
whoa I just implemented delete

**Comment by zcutlip** [comment]
I'm going to limit this enhancement to item and document deletion, and not include other object types (user, group, vault, etc). If there's a specific need for one or more of the others, please file a separate issue.

I don't think it'll be hard, but I'd like to keep the scope as narrow as possible if this is going to go out with the next release

**Comment by zcutlip** [comment]
I really want to get this release out the door, so I'm going to further reduce the scope of this issue to just "item deletion"

I'll create a new issue for documents. It'll be just as straightforward, but updating `pyonepassword`, and my `mock-op` testing tool, plus code documentation would take at least an evening but probably two or three

And after than we can add other object types such as users, groups, vaults, etc. on an issue-by-issue bases.

**Comment by zcutlip** [comment]
Here's the API for item `OP.item_delete()`. I'm about to merge it into `development` and it should be stable after that.

```python
    def item_delete(self, item_identifier: str, vault: str = None, archive: bool = False):
        """
        Delete an item based on title or unique identifier

        Parameters
        ----------
        item_identifier: str
            Any item identifier accepted by 'op item delete'
            Generally this includes item name/title, item ID, or share link
            See 'op item get --help' for more information
        vault: str, optional
            The name or ID of a vault to override the object's default vault
        archive: bool
            Whether to archive or permanently delete the item

        Note:
            If a non-unique item identifier is provided (e.g., item name/title), and there
            is more than one item that matches, OPItemGetException will be raised. Check the
            error message in OPItemGetException.err_output for details

        Raises
        ------
        OPItemDeleteException
            If the delete operation fails for any reason
        OPNotFoundException
            If the 1Password command can't be found
        Returns
        -------
        item_id: str
            Unique identifier of the item deleted

        """
        ...
```

**Comment by zcutlip** [comment]
Okay, we have item deletion as of #53

**Comment by zcutlip** [comment]
@nodomain, just wanted to update you that I've released v3.3.1, and it should be available in PyPI. It supports item creation and deletion as discussed previously

**Comment by nodomain** [comment]
Awesome! Currently booked with other projects but I’ll readdress this soon. Thank you very much.  Am 16.11.2022 um 06:45 schrieb Zachary Cutlip ***@***.***>:﻿
@nodomain, just wanted to update you that I've released v3.3.1, and it should be available in PyPI. It supports item creation and deletion as discussed previously

**Comment by zcutlip** [comment]
No worries! just wanted you to know! 🍻
